docs/source/features/passes/onnx.md

```diff
@@ -79,6 +79,103 @@ Please refer to [OrtTransformersOptimization](ort_transformers_optimization) for
+                ]
+            }
+        },
+        {
+            "CenterCrop": {
+                "height": {"type": "__model_input__", "input_index": 0, "dim_index": -2},
+                "width": {"type": "__model_input__", "input_index": 0, "dim_index": -1},
+            }
+        },
+        {"PixelsToYCbCr": {"layout": "BGR"}},
+        {"ImageBytesToFloat": {}},
+        {"Unsqueeze": {"axes": [0, 1]}},
+    ],
+    "post": [
+        {"Squeeze": {"axes": [0, 1]}},
+        {"FloatToImageBytes": {"name": "Y1_uint8"}},
+        {
+            "Resize": {
+                "params": {
+                    "resize_to": [
+                        {"type": "__model_output__", "output_index": 0, "dim_index": -2},
+                        {"type": "__model_output__", "output_index": 0, "dim_index": -1},
+                    ],
+                    "layout": "HW",
+                },
+                "io_map": [["PixelsToYCbCr", 1, 0]],
+            }
+        },
+        {"FloatToImageBytes": {"multiplier": 1.0, "name": "Cb1_uint8"}},
+        {
+            "Resize": {
+                "params": {
+                    "resize_to": [
+                        {"type": "__model_output__", "output_index": 0, "dim_index": -2},
+                        {"type": "__model_output__", "output_index": 0, "dim_index": -1},
+                    ],
+                    "layout": "HW",
+                },
+                "io_map": [["PixelsToYCbCr", 2, 0]],
+            }
+        },
+        {"FloatToImageBytes": {"multiplier": 1.0, "name": "Cr1_uint8"}},
+        {
+            "YCbCrToPixels": {
+                "params": {
+                    "layout": "BGR",
+                },
+                "io_map": [
+                    ["Y1_uint8", 0, 0],
+                    ["Cb1_uint8", 0, 1],
+                    ["Cr1_uint8", 0, 2],
+                ],
+            }
+        },
+        {"ConvertBGRToImage": {"image_format": "png"}},
+    ],
+    "tool_command_args": [
+        {
+            "name": "image",
+            "data_type": "uint8",
+            "shape": ["num_bytes"],
+        }
+    ],
+    "target_opset": 16,
+}
+```
+
+* Olive introduce two placeholders to represent the model input/ouput shape dimension value: `__model_input__` and `__model_output__`.
+* To support the IoMapEntry, the step need choose use the full form. For example:
+```json
+    "YCbCrToPixels": {
+        "params": {
+            "layout": "BGR",
+        },
+        "io_map": [
+            ["Y1_uint8", 0, 0],
+            ["Cb1_uint8", 0, 1],
+            ["Cr1_uint8", 0, 2],
+        ],
+    }
+```
+* The `tool_command_args` will be used to describe the input parameters to create the `PrePostProcessor` instance. It is list of `PrePostProcessorInput`.
+  The `name` is the tensor name. The `data_type` and `shape` will be used to create the tensor type. The `shape` can be a list of integers or a list of string.
```

Comment:
What is the difference for list[int] vs list[str] for `shape`?

Reply:
it is used by PrePostProcessor  when creating the inputs. Here is how the shape is used. https://github.com/onnx/onnx/blob/main/onnx/helper.py#L1022-L1025